File: 프로그래머스/lv2.프린터.js
// Level2: 프린터
function solution(priorities, location) {
    let answer = 0;
    let queue = priorities.map((p,idx) => [p,idx]);
    while(true){
        let element = queue.shift();
        if(queue.some(m => m[0] > element[0])){
            queue.push(element);
        }
        else{
            answer += 1;
            if (element[1] === location){
                return answer;
            }
        }
    }
}

File: 프로그래머스/lv2.프린터.py
# Level2: 프린터
from collections import deque
def solution(priorities, location):
    answer = 0
    queue = deque()
    for idx, priority in enumerate(priorities):
        queue.append([priority,idx])
    while queue:
        priority, idx = queue.popleft()
        if queue and max(queue)[0] > priority:
            queue.append([priority,idx])
        else:
            answer += 1
            if location == idx:
                return answer
